services/controllerhost/dfdd.go

```diff
@@ -58,246 +62,292 @@ type (
 		inputListenerCh chan *common.RingpopListenerEvent
 		storeListenerCh chan *common.RingpopListenerEvent
 
-		healthCheckTicker   *time.Ticker
-		healthCheckInterval time.Duration
-
-		unhealthyStores     map[string]time.Time
-		unhealthyStoresLock sync.RWMutex
+		inputHosts atomic.Value
+		storeHosts atomic.Value
 
-		unhealthyInputs     map[string]time.Time
-		unhealthyInputsLock sync.RWMutex
-
-		hostDownPeriodForStage2 time.Duration
+		timeSource common.TimeSource
 	}
 )
 
-const (
-	inputServiceID serviceID = iota
-	outputServiceID
-	storeServiceID
-)
+var errUnknownService = errors.New("dfdd: unknown service")
 
 const (
 	listenerChannelSize = 32
-
-	healthCheckInterval = time.Duration(1 * time.Minute)
 )
 
+type dfddHostState int
+
+/*
+ * State Transitions
+ *
+ *        Unknown  -- rp.HostAddedEvent           --> UP
+ *             UP  -- loadReporter.HostGoingDown  --> GoingDown
+ *             UP  -- rp.HostRemovedEvent         --> Down
+ *      GoingDown  -- rp.HostRemovedEvent         --> Down
```

Comment:
can you add comment somewhere about the difference between input and store(i.e. input will removed immediately once down, while store will have the down/forgotten transition)?